### ISSUE TITLE
Disconnect Editor Rendering MSAA Quality Level From UserData

### ISSUE THREAD

**Issue by ilexp** [AdamsLair/duality#548]
### Summary

Right now, the editor will use an MSAA sample count based on the default user data of the game for its rendering site backbuffers. Instead, use a per-site quality level that is disconnected from the game itself.

### Analysis

- Prerequisite: Issue #515 
- Adjust the editor backend to allow specifying antialiasing quality on a per-site basis and remove the code that derives it from user data.
- Make sure RenderTarget antialiasing quality remains unaffected by this change.
- Adjust the Game View to match user data antialiasing quality and settings. If that quality level is not equal to the one used by the camera view, this will force using an offscreen buffer for game rendering.
- Figure out a good default for editor view rendering AA quality levels. Could be "High" for now, to match the user data defaults.

**Comment by ilexp** [comment]
Postponing this to Future milestone, as it's completely non-critical and its details should probably be considered with the bigger picture in mind.

**Comment by ilexp** [comment]
Another difference to be considered is [texture memory usage on high MSAA levels](https://mynameismjp.wordpress.com/2012/10/24/msaa-overview/).

**Comment by ilexp** [comment]
Putting this back into v3.0, as the higher level considerations that made me move this to Future can actually be postponed independently by simply using UserData AA quality as the default for every rendering site instance. That way, behavior won't change at all, but the API will be prepared.

**Comment by ilexp** [comment]
Done in [this commit](https://github.com/AdamsLair/duality/commit/4b8a77df48a2c13e6d2fc1105702249f4f071c3f). Note that it is not possible to share the same OpenGL context across multiple windows with different graphics modes, so editor antialiasing quality is now a setting that is specified on creating an `INativeEditorGraphicsContext` rather than on creating individual `INativeRenderableSite` instances. The Game View checks for its sites AA quality level and uses an offscreen buffer for rendering if that quality level is different from the games target quality level.

Closing this.
